x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/ExponentialHistogramBlockBuilder.java

```diff
@@ -115,28 +125,83 @@ public ExponentialHistogramBlockBuilder append(ExponentialHistogram histogram) {
         } catch (IOException e) {
             throw new RuntimeException("Failed to encode histogram", e);
         }
-        if (Double.isNaN(histogram.min())) {
+        double sum;
+        if (histogram.valueCount() == 0) {
+            assert histogram.sum() == 0.0 : "Empty histogram should have sum 0.0 but was " + histogram.sum();
+            sum = Double.NaN; // we store null/NaN for empty histograms to ensure avg is null/0.0 instead of 0.0/0.0
+        } else {
+            sum = histogram.sum();
+        }
+        return new EncodedHistogramData(
+            histogram.valueCount(),
+            sum,
+            histogram.min(),
+            histogram.max(),
+            zeroBucket.zeroThreshold(),
+            encodedBytes.bytes().toBytesRef()
+        );
+    }
+
+    public ExponentialHistogramBlockBuilder append(ExponentialHistogram histogram) {
+        EncodedHistogramData data = encode(histogram);
+        valueCountsBuilder.appendDouble(data.count);
+        if (Double.isNaN(data.min)) {
             minimaBuilder.appendNull();
         } else {
-            minimaBuilder.appendDouble(histogram.min());
+            minimaBuilder.appendDouble(data.min);
         }
-        if (Double.isNaN(histogram.max())) {
+        if (Double.isNaN(data.max)) {
             maximaBuilder.appendNull();
         } else {
-            maximaBuilder.appendDouble(histogram.max());
+            maximaBuilder.appendDouble(data.max);
         }
-        if (histogram.valueCount() == 0) {
-            assert histogram.sum() == 0.0 : "Empty histogram should have sum 0.0 but was " + histogram.sum();
+        if (Double.isNaN(data.sum)) {
             sumsBuilder.appendNull();
         } else {
-            sumsBuilder.appendDouble(histogram.sum());
+            sumsBuilder.appendDouble(data.sum);
         }
-        valueCountsBuilder.appendDouble(histogram.valueCount());
-        zeroThresholdsBuilder.appendDouble(zeroBucket.zeroThreshold());
-        encodedHistogramsBuilder.appendBytesRef(encodedBytes.bytes().toBytesRef());
+        zeroThresholdsBuilder.appendDouble(data.zeroThreshold);
+        encodedHistogramsBuilder.appendBytesRef(data.encodedHistogram);
         return this;
     }
 
+    static ExponentialHistogramBlock createConstant(ExponentialHistogram histogram, int positionCount, BlockFactory blockFactory) {
```

Comment:
nit: should we move this method to ExponentialHistogramArrayBlock not in this builder?